src/main.c

```diff
@@ -2438,7 +2463,7 @@ void handle_movement(double dt) {
             }
         }
     }
-    float speed = g->flying ? 20 : 5;
+    float speed = g->flying ? 20 : *movement_speed_ptr;
```

Comment:
I do not see a good reason for movement_speed_ptr being a pointer instead of a local variable inside of the function. I think it would make more sense for the function to determine the speed based on the current state instead of retaining speed between each time handle_movement is called.

Reply:
The reason I am using a pointer instead of a literal value. Is because the handle_movement() function used to use a literal for the running speed outside of the functionality I implemented. The problem with that, is when the handle_movement() function gets called in quick succession which it does. If I instead decided to make a local variable within said function, then the variable would be reset every time handle_movement() is called of course. Meaning if the player pressed the "running" key to increase movement speed to 10, once handle_movement() is called again then the value of the running speed will be reset to the default local variable value via variable declaration. 
Making a pointer & using it in the way I did, guarantees that once the speed is changed, it stays changed until I actually want it not changed because the craft_main() function is ran once, unlike handle_movement().